src/legacy/core_plugins/kibana/public/management/sections/settings/index.js

```diff
@@ -82,13 +83,18 @@ uiRoutes.when('/management/kibana/settings/:setting?', {
   },
 });
 
-uiModules.get('apps/management').directive('kbnManagementAdvanced', function(config, $route) {
+uiModules.get('apps/management').directive('kbnManagementAdvanced', function($route) {
   return {
     restrict: 'E',
     link: function($scope) {
-      config.watchAll(() => {
-        updateAdvancedSettings($scope, config, $route.current.params.setting || '');
-      }, $scope);
+      updateAdvancedSettings($scope, npStart.core.uiSettings, $route.current.params.setting || '');
+      npStart.core.uiSettings.getUpdate$().subscribe(() => {
```

Comment:
Changes in this file cause a bug in the functionality (if the changes are reverted, all works as expected).
Steps to reproduce:
1. Load Kibana and go to advanced settings
2. Modify a chosen setting.
3. Click on 'save'
5. Actual behaviour: the setting is being saved, the network request `api/settings` is run BUT *the save button and cancel button stay in EUI - looks like UI components are not notified about the first change.

4. Expected behaviour: the setting is being saved, the network request is run and the save button and cancel button disappears

It only doesn't work for the first attempt of saving, after that all works fine.

Reply:
Thanks for bringing this to my attention - fixed!